components/sup/src/manager/service.rs

```diff
@@ -921,11 +934,23 @@ impl Service {
         win_perm::harden_path(path.as_ref())
     }
 
+    /// Called when the Supervisor reattaches itself to an already
+    /// running service. Use this to re-initiate any associated
+    /// processes, futures, etc.
+    ///
+    /// This should generally be the opposite of `Service::detatch`.
+    fn reattach(&mut self, executor: &TaskExecutor) {
+        outputln!("Reattaching to {}", self.service_group);
+        self.initialized = true;
+        self.restart_health_checks(executor);
+    }
+
     fn execute_hooks(&mut self, launcher: &LauncherCli, executor: &TaskExecutor) {
         if !self.initialized {
             if self.check_process() {
-                outputln!("Reattached to {}", self.service_group);
-                self.initialized = true;
+                // TODO (CM): seems odd for "reattach" logic to live
+                // in `execute_hooks`... consider how to refactor.
+                self.reattach(executor);
```

Comment:
I'd suggest moving this logic _inside_ `initialize` so that the management of the state is kept localized

Reply:
I'll investigate this (and your other suggestion) as I dig into this `TODO` item.